### PR TITLE
Clean up unit-test invocation with tox / Travis CI

### PR DESCRIPTION
**Pull request by adamhadani**
Addresses most issues raised in #19 -
1. Switch to use tox for running unit-tests in Travis CI, removing redundant unit-test running code definitions from Travis configuration file
2. Adds flake8 (linting) target to the tox configuration for linting code and running style checks as part of CI
3. Remove unneeded requirements-py{2,3}.txt file that were previously used by travis CI config
4. adds default flake8 configuration to setup.cfg

~~These changes need to be checked once this PR is live and Travis CI actually picks up the branch and tries to build it, might need a couple of tweaks to get it fully working, will update this PR with comment once looks good (and should be able to see once Travis CI checks pass)~~
all tests verified to pass on Travis CI: https://travis-ci.org/RDFLib/rdflib-sqlalchemy/builds/162038128

### CODE REVIEW

File: test/test_aggregate_graphs.py [hunk 1]
```diff
@@ -127,7 +127,6 @@ class GraphAggregates2(unittest.TestCase):
     def setUp(self):
         memStore = plugin.get('SQLAlchemy', Store)(
             identifier="rdflib_test", configuration=Literal("sqlite://"))
-        print("ASDF")
```
Comment (adamhadani) [inline test/test_aggregate_graphs.py]: left those dangling prints in my previous PR, caught them here with help of flake8-print that now runs as part of CI!

File: tox.ini [hunk 2]
```diff
@@ -10,7 +10,7 @@ commands =
 deps =
     nose
     psycopg2
-    MySQL-python
+    mysqlclient
```
Comment (adamhadani) [inline tox.ini]: we were previously using 'mysqlclient' for some python versions (3.x) using direct git repository URL (see below), and for 2.x using the old MySQL-python package. Switched to just use the 'mysqlclient' fork (which is same as the one pointed to via the GitHub URL below) across the board - it seems very actively maintained and compliant with both python 2.x and 3.x - https://github.com/PyMySQL/mysqlclient-python

File: .travis.yml [hunk 3]
```diff
@@ -15,17 +15,9 @@ python:
     - 3.4
     - 3.5
 install:
-    - if [[ ${TRAVIS_PYTHON_VERSION%%.*} == '2' ]]; then pip install -r requirements-py2.txt; fi
-    - if [[ ${TRAVIS_PYTHON_VERSION%%.*} == '3' ]]; then pip install -r requirements-py3.txt; fi
-    - python setup.py install
```
Comment (adamhadani) [inline .travis.yml]: no need to explicitly install package or dependencies here anymore - tox manages this as part of its configuration / runtime. only required dependencies here are immediate bootstrap packages for running tox and installing dependencies later (setuptools)

File: .travis.yml [hunk 4]
```diff
@@ -1,9 +1,5 @@
 # http://travis-ci.org/#!/gjhiggins/rdflib-sqlalchemy
 language: python
-branches:
```
Comment (adamhadani) [inline .travis.yml]: don't think we should be limiting when to run CI, esp. if we adopt something like the git-flow model that uses feature branches more extensively. It's possible to exclude certain branches to form a 'blacklist' which might be useful in some cases

File: tox.ini [hunk 5]
```diff
@@ -46,11 +53,8 @@ deps =
...
-    MySQL-python
-
-# we separate coverage into its own testenv because a) "last run wins" wrt
-# cobertura jenkins reporting and b) pypy and jython can't handle any
-# combination of versions of coverage and nosexcover that i can find.
-# coverage <3.4 is required by nosexcover 1.0.4.
+    mysqlclient
 
-# DB='pgsql' DBURI='postgresql+psycopg2://user:password@host/dbname' tox -e py32
+[tox:travis]
```
Comment (adamhadani) [inline tox.ini]: this is required for invoking the cover and lint targets in addition to the database/python version build matrix.
I've elected to only run them on the 2.7 and 3.5 targets and keep coverage to 2.7 (as it was before limited to use python 2.7)

**Comment by adamhadani** [comment]
@gjhiggins LMK if looks good to you I'll go ahead and merge this, verified all tests pass

**Comment by adamhadani** [comment]
planning to merge this in unless I hear differently in the next couple of days, cc/ @gjhiggins

**Comment by gjhiggins** [comment]
Go for it :)

**Pull request merged**